src/ui_elements/menu_base.gd

```diff
@@ -0,0 +1,25 @@
+extends Control
+
+# --Private Functions--
+
+func _ready():
+	$Darken.hide()
+	if Scenes.baseScene == null:
+		Scenes.setBase(self, "res://ui_elements/main_menu.tscn")
```

Comment:
...here.
I mean, here the setBase and the overlay functions could be called after each other, instead of the setBase needing a path to call overlay on.